search-solr/src/main/java/edu/unc/lib/boxc/search/solr/services/SearchStateFactory.java

```diff
@@ -397,13 +397,35 @@ private void populateSearchState(SearchState searchState, Map<String,String[]> r
                 }
             }
             searchState.setFacetsToRetrieve(facetsToRetrieve);
+
+
+            //Add date added year range, which is a sort of facet
+            try {
+                SearchState.RangePair dateCreatedYear = new SearchState.RangePair();
+                parameter = getParameter(request, searchSettings.searchFieldParam(
+                        SearchFieldKey.DATE_CREATED_YEAR.name()));
+                if (parameter != null && parameter.length() > 0) {
+                    String[] dateRange = parameter.split(",");
+                    dateCreatedYear.setLeftHand(dateRange[0]);
+                    dateCreatedYear.setRightHand(dateRange[1]);
+                }
+
+                if (dateCreatedYear.getLeftHand() != null && dateCreatedYear.getRightHand() != null) {
+                    searchState.getRangeFields().put(SearchFieldKey.DATE_CREATED_YEAR.name(), dateCreatedYear);
+                }
+            } catch (IllegalArgumentException e) {
+                // An invalid range was specified, throw away the range pair
+                Map<String, SearchState.RangePair> currentRangeFields = searchState.getRangeFields();
```

Comment:
Did it ever occur that the RangePair got left behind after an exception was thrown? It seems like it shouldn't, since it doesn't get added to searchState.rangeFields until after validation would have occurred. So might be able to remove this part (it's a good idea to keep a log statement about the error having occurred though)

Reply:
Yes, it seems to be there. This threw me off for a good while, as I was thinking the same thing. So, I ended up explicitly removing it.